src/main/java/shoppingcart/service/OrderService.java

```diff
@@ -1,19 +1,17 @@
 package shoppingcart.service;
 
-import java.util.ArrayList;
-import java.util.List;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
-import shoppingcart.dao.CartItemDao;
-import shoppingcart.dao.CustomerDao;
-import shoppingcart.dao.OrderDao;
-import shoppingcart.dao.OrdersDetailDao;
-import shoppingcart.dao.ProductDao;
-import shoppingcart.dto.OrderDetailRequestDto;
-import shoppingcart.dto.OrderDetailResponseDto;
-import shoppingcart.dto.OrderResponseDto;
-import shoppingcart.dto.OrdersDetail;
-import shoppingcart.dto.Product;
+import shoppingcart.dao.*;
+import shoppingcart.dto.OrderDetailDto;
+import shoppingcart.dto.OrderRequestDto;
+import shoppingcart.dto.OrdersDto;
+import shoppingcart.dto.ProductDto;
+import shoppingcart.exception.InvalidOrderException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
 @Transactional
```

Comment:
```suggestion
@Transactional(rollbackFor = Exception.class)
```

지금 크게 문제는 없을꺼 같지만, 만약 예외가 발생해서 모든 과정이 롤백되길 원한다면 현재 예상되는 예외들을 checked 해서 처리하고 있기 때문에 서비스 단의 `@Transactional`에 옵션을 넣어주면 좋을것 같네요.

[Transactional 디폴트 옵션은 Checked Exception을 롤백하지 않는다](https://interconnection.tistory.com/122)

Reply:
제쪽에서 수정했기 때문에 리졸브 해놓겠습니다.